From 1d9c97edb2d77913c263fcbd23b6efc50ada454c Mon Sep 17 00:00:00 2001
From: AtifChy <iftakhar.awal@yahoo.com>
Date: Sun, 8 Aug 2021 20:44:22 +0600
Subject: [PATCH 05/17] dynamic-cursor-v2

---
 x.c | 24 +++++++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/x.c b/x.c
index d24a66c..9887a97 100644
--- a/x.c
+++ b/x.c
@@ -1631,6 +1631,7 @@ void
 xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og, Line line, int len)
 {
 	Color drawcol;
+	XRenderColor colbg;
 
 	/* remove the old cursor */
 	if (selected(ox, oy))
@@ -1663,10 +1664,27 @@ xdrawcursor(int cx, int cy, Glyph g, int ox, int oy, Glyph og, Line line, int le
 			g.fg = defaultfg;
 			g.bg = defaultrcs;
 		} else {
-			g.fg = defaultbg;
-			g.bg = defaultcs;
+			/** this is the main part of the dynamic cursor color patch */
+			if (!(og.mode & ATTR_REVERSE)) {
+				unsigned long col = g.bg;
+				g.bg = g.fg;
+				g.fg = col;
+			}
+		}
+
+		/**
+		 * and this is the second part of the dynamic cursor color patch.
+		 * it handles the `drawcol` variable
+		*/
+		if (IS_TRUECOL(g.bg)) {
+			colbg.alpha = 0xffff;
+			colbg.red = TRUERED(g.bg);
+			colbg.green = TRUEGREEN(g.bg);
+			colbg.blue = TRUEBLUE(g.bg);
+			XftColorAllocValue(xw.dpy, xw.vis, xw.cmap, &colbg, &drawcol);
+		} else {
+			drawcol = dc.col[g.bg];
 		}
-		drawcol = dc.col[g.bg];
 	}
 
 	/* draw the new one */
-- 
2.32.0

