From a5b83fe56dd1ceac35cca5b1627f2025a30c60b7 Mon Sep 17 00:00:00 2001
From: AtifChy <iftakhar.awal@yahoo.com>
Date: Mon, 21 Jun 2021 23:18:33 +0600
Subject: [PATCH 07/13] clipboard

---
 config.def.h | 6 +++---
 x.c          | 2 ++
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/config.def.h b/config.def.h
index 8135b56..cf1f387 100644
--- a/config.def.h
+++ b/config.def.h
@@ -230,7 +230,7 @@ ResourcePref resources[] = {
  */
 static MouseShortcut mshortcuts[] = {
 	/* mask                 button   function        argument       release */
-	{ XK_ANY_MOD,           Button2, selpaste,       {.i = 0},      1 },
+	{ XK_ANY_MOD,           Button2, clippaste,      {.i = 0},      1 },
 	{ ShiftMask,            Button4, ttysend,        {.s = "\033[5;2~"} },
 	{ XK_ANY_MOD,           Button4, ttysend,        {.s = "\031"} },
 	{ ShiftMask,            Button5, ttysend,        {.s = "\033[6;2~"} },
@@ -252,8 +252,8 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Home,        zoomreset,      {.f =  0} },
 	{ TERMMOD,              XK_C,           clipcopy,       {.i =  0} },
 	{ TERMMOD,              XK_V,           clippaste,      {.i =  0} },
-	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
-	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
+	{ TERMMOD,              XK_Y,           clippaste,      {.i =  0} },
+	{ ShiftMask,            XK_Insert,      clippaste,      {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
 };

diff --git a/x.c b/x.c
index aa07260..a2e335c 100644
--- a/x.c
+++ b/x.c
@@ -688,6 +688,8 @@ setsel(char *str, Time t)
 	XSetSelectionOwner(xw.dpy, XA_PRIMARY, xw.win, t);
 	if (XGetSelectionOwner(xw.dpy, XA_PRIMARY) != xw.win)
 		selclear();
+
+	xclipcopy();
 }

 void
--
2.32.0

